src/backend/cpu/platform.cpp

```diff
@@ -249,43 +169,50 @@ std::string getDeviceInfo()
     info << AF_COMPILER_STR;
 #endif
     info << std::endl;
+
     return info.str();
 }
 
 bool isDoubleSupported(int device)
 {
-    return true;
+    return DeviceManager::IS_DOUBLE_SUPPORTED;
```

Comment:
Just moved all(possible) hard coded constants into single location(static data members of DeviceManager class).